react/players/VimeoPlayer.tsx

```diff
@@ -0,0 +1,41 @@
+import React, { FunctionComponent } from 'react'
+import { useCssHandles } from 'vtex.css-handles'
+
+import { VideoPlayer } from '../interfaces'
+
+const CSS_HANDLES = ['videoContainer', 'videoElement'] as const
+
+// https://regex101.com/r/wdKKHO/1
+const VIMEO_REGEX = /^.*(vimeo\.com\/)((channels\/[A-z]+\/)|(groups\/[A-z]+\/videos\/))?([0-9]+)/
+
+const VimeoPlayer: FunctionComponent<VideoPlayer> = ({
+  width,
+  height,
+  autoPlay,
+  loop,
+  src,
+  description,
+}) => {
+  const handles = useCssHandles(CSS_HANDLES)
+  const params = `autoplay=${autoPlay}&loop=${loop}&enablejsapi=1&iv_load_policy=3&modestbranding=1`
+  const matchedSrc = src.match(VIMEO_REGEX)
+  const videoId = matchedSrc?.[5]
```

Comment:
If the other capturing groups are irrelevant, you can the non-capturing group modifier `?:` to prevent having to get the fifth index instead of the first.